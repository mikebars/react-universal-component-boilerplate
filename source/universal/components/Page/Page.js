/* @flow */

import styled from 'styled-components'

export const Page = styled.section`
  background-color: ${props => props.backgroundColor};

  height: 100%;

  margin: 0;

  padding: 0;

  width: 100%;
`

export default Page
