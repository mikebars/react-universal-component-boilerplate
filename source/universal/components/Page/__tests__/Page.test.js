/* @flow */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount, shallow, render } from 'enzyme'

import { Page } from 'universal/components/Page/Page'

test('Page - snapshot test', () => {
  const props = {}

  const element = <Page {...props} />

  const tree = renderer.create(element).toJSON()

  expect(tree).toMatchSnapshot()
})

test('Page - enzyme shallow render', () => {
  const props = {}

  const element = <Page {...props} />

  const wrapper = shallow(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})

test('Page - enzyme full render', () => {
  const props = {}

  const element = <Page {...props} />

  const wrapper = mount(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})

test('Page - enzyme static render', () => {
  const props = {}

  const element = <Page {...props} />

  const wrapper = render(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})
