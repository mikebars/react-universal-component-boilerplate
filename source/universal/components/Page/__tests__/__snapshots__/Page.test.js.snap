// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page - snapshot test 1`] = `
<section
  className="Page-o0cu63-0 dlJkxw"
/>
`;
